### PR TITLE
Task 4 get article

### PR DESCRIPTION
**Pull request by JoeB025**
Created a request to get article by id 

created tests to check the correct information from the request id comes through 

created error tests to check for any bad requests and requests for invalid endpoints.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -69,4 +64,65 @@ describe('app', () => {
...
+      .then((res) => {
+       expect(res.body.article.title).toBe('Living in the shadow of a great man')
+       expect(res.body.article.topic).toBe('mitch')
+       expect(res.body.article.author).toBe('butter_bridge')
+       expect(res.body.article.body).toBe('I find this existence challenging')
+       expect(res.body.article.created_at).toBe('2020-07-09T20:11:00.000Z')
+       expect(res.body.article.votes).toBe(100)
+       expect(typeof res.body.article.votes).toBe('number')
+       expect(res.body.article.article_img_url).toBe('https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700');
+          })
```
Comment (nc-roisin-oreilly) [inline __tests__/app.test.js:89]: nicely done

File: __tests__/app.test.js [hunk 2]
```diff
@@ -69,4 +64,65 @@ describe('app', () => {
...
+          expect(res.body.article.body).toBe('Bastet walks amongst us, and the cats are taking arms!');
+          })
+        })
+    test('GET /articles should return a status code 404 with the message endpoint not found', () => {
+      return request(app)
+        .get('/api/no-articles/1') 
+        .expect(404)
+        .then((res) => {
+          expect(res.body).toEqual({Status: 404, msg: 'endpoint not found'})
+          expect(res.body.msg).toBe('endpoint not found')
```
Comment (nc-roisin-oreilly) [inline __tests__/app.test.js:109]: I would just keep this line and delete the previous one

File: models/articles.model.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+const db = require('../db/connection')
+
+
+exports.selectArticles = (article_id) => {
+  return db
+    .query('SELECT * FROM articles WHERE article_id=$1;', [article_id])
+    .then((result) => {
+      if (result.rows.length === 0) {
+        return Promise.reject({
+          msg: 'article does not exist'
```
Comment (nc-roisin-oreilly) [inline models/articles.model.js:10]: you can reject with a status code here too

**Review by nc-roisin-oreilly — COMMENTED** [review]
really great work :) I've left a few small comments but PR approved, keep it up!

File: __tests__/app.test.js (hunk 2 above)
Comment (JoeB025) [reply __tests__/app.test.js:109]: as in remove line 108? :)

**Pull request merged**